### PR TITLE
IBX-5190 Missing object types in REST API reference

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-5190
| Versions      | all
| Edition       | 4.4

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 1]
```diff
@@ -2119,7 +2123,15 @@ VersionTranslationInfo:
             type: LanguageCode[]
 
 VersionUpdate:
+    description: 'This class is used to update a version.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2126]:
```suggestion
    description: 'This class is used to update a content version.'
```
?

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 2]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+        remoteId:
+            type: string
+        urlAliasSchema:
+            type: string
+        nameSchema:
+            type: string
+        isContainer:
+            type: boolean
+        defaultSortField:
+            description: 'Specifies which property the child locations should be sorted on by default when created.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2169]:
```suggestion
            description: 'Specifies which property the child Locations should be sorted on by default when created.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 3]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+        defaultAlwaysAvailable:
+            type: boolean
+        names:
+            type: array
+        descriptions:
+            type: array
+        FieldDefinition:
+            type: FieldDefinition
+        creatorId:
+            description: 'If set this value overrides the current user as creator.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2183]:
```suggestion
            description: 'If set, this value overrides the current user as creator.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 4]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+        remoteId:
+            type: string
+        urlAliasSchema:
+            type: string
+        nameSchema:
+            type: string
+        isContainer:
+            type: boolean
+        defaultSortField:
+            description: 'Specifies which property the child locations should be sorted on by default when updated.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2206]:
```suggestion
            description: 'Specifies which property the child Locations should be sorted on by default when updated.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 5]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+            description: 'Specifies whether the sort order should be ascending or descending by default when updated.'
+            type: mixed
+        defaultAlwaysAvailable:
+            type: boolean
+        names:
+            type: array
+        descriptions:
+            type: array
+        modifierId:
+            description: 'If set this value overrides the current user as creator.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2218]:
```suggestion
            description: 'If set, this value overrides the current user as creator.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 6]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+            type: string
+        fields:
+            type: Field[]
 
 RoleAssignInput:
+    description: 'This class represents a Role assign input.'
     type: BaseObject
+    properties:
+        id:
+            description: 'The unique ID of the role assignment.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2266]:
```suggestion
            description: 'The unique ID of the Role assignment.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 7]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+            type: integer
+        Role:
+            description: 'Returns the role to which the user or user group is assigned to.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2269]:
```suggestion
            description: 'Returns the Role to which the user or user group is assigned to.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 8]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+            type: Role
+        Limitation:
+            description: 'Returns the limitation of the role assignment.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2272]:
```suggestion
            description: 'Returns the Limitation of the Role assignment.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 9]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+            type: string
+        enabled:
+            type: boolean
+        maxLogin:
+            type: integer
+        ContentUpdate:
+            description: 'The update structure for the profile content.'
+            type: ContentUpdate
+        ContentMetadataUpdate:
+            description: 'The update structure for the profile meta data.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2293]:
```suggestion
            description: 'The update structure for the profile metadata.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 10]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+    properties:
+        identifier:
+            type: string
 
 PolicyCreate:
+    description: 'This class is used to create a Policy.'
     type: BaseObject
+    properties:
+        module:
+            description: 'Name of module, associated with the Policy. For example, content.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2308]:
```suggestion
            description: 'Name of module associated with the Policy. For example, content.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 11]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+            type: string
+        function:
+            description: 'Name of the module function Or all functions with '*'. For example, read.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2311]:
```suggestion
            description: 'Name of the module function, or all functions with '*'. For example, read.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 12]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+    properties:
+        name:
+            type: string
+        identifier:
+            type: string
+        position:
+            type: integer
 
 AttributeGroupCreate:
+    description: 'This class is used to creat an attribute group.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2606]:
```suggestion
    description: 'This class is used to create an attribute group.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 13]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
 ProductTypeUsage:
+    description: 'This class represents a product type usage.'
     type: object
-    description: ''
+    properties:
+        isUsed:
+            type: boolean
 
 ProductTypeCreate:
+    description: 'This class is used to creat a product type.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2847]:
```suggestion
    description: 'This class is used to create a product type.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 14]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
 CurrencyViewInput:
+    description: 'This class represents a currency view input.'
     type: object
-    description: ''
+    properties:
+        CurrencyQuery:
+            type: object
 
 ProductType:
+    description: 'This class represents a Product type.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2829]:
```suggestion
    description: 'This class represents a product type.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 15]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
     type: object
-    description: ''
+    properties:
+        content:
+            type: integer
+        entries:
+            type: TaxonomyEntry[]
 
 TaxonomyEntryUnassignFromContent:
+    description: 'This class is used to unassign taxonomy entry to Content.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:3170]:
```suggestion
    description: 'This class is used to unassign taxonomy entry from content.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 16]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
 
 TaxonomyEntryBulkMove:
+    description: 'This class is used to bulk move taxonomy entries.'
     type: object
+    properties:
+        entries:
+            type: TaxonomyEntryMove[]
 
 TaxonomyEntryAssignToContent:
+    description: 'This class is used to assign taxonomy entry to Content.'
```
Comment (DominikaK) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:3161]:
```suggestion
    description: 'This class is used to assign taxonomy entry to content.'
```

**Review by DominikaK — APPROVED** [review]

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 17]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
 
 PolicyCreate:
+    description: 'This class is used to create a Policy.'
     type: BaseObject
+    properties:
+        module:
+            description: 'Name of module associated with the Policy. For example, content.'
+            type: string
+        function:
+            description: 'Name of the module function, or all functions with '*'. For example, read.'
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2311]: `Unexpected characters near "*'. For example, read.'" at line 2311 (near "description: 'Name of the module function, or all functions with '*'. For example, read.'").`

In YAML (and RAML by extension), to escape a single quote, just double it:

```suggestion
            description: 'Name of the module function, or all functions with ''*''. For example, read.'
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 18]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+            type: Limitation[]
 
 SessionInput:
+    description: 'This class represents a session input.'
     type: BaseObject
+        properties:
+            login:
+                type: string
+            password:
+                type: string
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2330]: `A colon cannot be used in an unquoted mapping value at line 2330 (near "            type: string").`

Just fix indent error:

```suggestion
    properties:
        login:
            type: string
        password:
            type: string
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 19]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+             type: string
+        name:
+            type: string
 
 AttributeTypeList:
+    description: 'This class represents an attribute type list.'
     type: object
-    description: ''
+    properties:
+        attributeTypes
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2747]: `Mapping values are not allowed in multi-line blocks at line 2747 (near "attributeTypes").`

Missing colon:

```suggestion
        attributeTypes:
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 20]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+                    type: boolean
+
 
 ProductTypeViewInput:
+    description: 'This class represents a product type view input.'
     type: object
-    description: ''
+        properties:
+            ProductTypeQuery:
+                type: object
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2886]: `A colon cannot be used in an unquoted mapping value at line 2886 (near "            type: object").`

Indent error fix:
```suggestion
    properties:
        ProductTypeQuery:
            type: object
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 21]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+        Entry:
+            type: Entry
 
 EntryAssignmentCollection:
+    description: 'This class represents a taxonomy assignment collection.'
     type: object
+    properties:
+        Content:
+            type: Content
+        assignments
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:3111]: `Unable to parse at line 3111 (near "assignments").`

Missing colon:
```suggestion
        assignments:
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 22]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+        identifier:
+            type: string
+        taxonomy:
+            type: string
+        Parent:
+            type: Parent
+        Content:
+            type: Content
+
+TaxonomyEntryUpdateStruct
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:3127]: `Unable to parse at line 3127 (near "TaxonomyEntryUpdateStruct").`

Missing colon:
```suggestion
TaxonomyEntryUpdateStruct:
```

**Review by adriendupuis — CHANGES_REQUESTED** [review]
There is few YAML/RAML syntax errors, see suggestions.

The type `mixed` isn't supported. When the type can't be more precise, [`any`](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#the-any-type) is the RAML keyword to use.

The type [`array`](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#array-type) should be precised with its items type. It will become mandatory with #1871
There is two syntaxes:
```yaml
type: aType[]
```
```yaml
type: array
items: aType
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 23]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+        identifier:
+            type: string
+        name:
+            type: string
+        mainLanguageCode:
+            type: string
+        names:
+            type: array
+        Parent:
+            type: Parent
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:3086]: `No type found for name 'Parent'`

```suggestion
            type: TaxonomyEntry
```

Probably the same thing for `TaxonomyEntryCreateStruct` and `TaxonomyEntryUpdateStruct`

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 24]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
 TaxonomyEntryAssignment:
+    description: 'This class represents a taxonomy entry assignment.'
     type: object
+    properties:
+        id:
+            type: integer
+        Content:
+            type: Content
+        Entry:
+            type: Entry
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:3103]: `No type found for name 'Entry'`

```suggestion
            type: TaxonomyEntry
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 25]
```diff
@@ -2128,401 +2140,1056 @@ View: #TODO outdated example
...
+            type: string
+        identifier:
+             type: string
+        description:
+             type: string
+        users:
+            description: 'A list of users attached to the customer group.'
+            type: User[]
+        global_price_rate:
+            type: numeric-string
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2766]: `No type found for name 'numeric-string'`

```suggestion
            description: 'A numeric string'
            type: string
```

**Comment by adriendupuis** [comment]
After applying localy all my suggestions, I still have to declare few more types at the end, quick and dirty:

```yaml
DateTime:
    description: https://www.php.net/manual/en/class.datetime.php
    type: object

ContentUpdate:
    type: object

ContentMetadataUpdate:
    type: object

SummaryEntryCollection:
    type: object

Price:
    type: object

VatCategorySummary:
    type: object

Thumbnail:
    type: object

AttributesList:
    type: object
```

File: docs/api/rest_api/rest_api_reference/input/ez-types.raml [hunk 26]
```diff
@@ -2128,401 +2141,1153 @@ View: #TODO outdated example
...
+        options:
+            type: array
 
 AttributeUpdate:
+    description: 'This class is used to update an attribute.'
     type: object
-    description: ''
+    properties:
+        identifier:
+             type: string
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-types.raml:2698]:
```suggestion
            type: string
```

Comment (adriendupuis) [reply docs/api/rest_api/rest_api_reference/input/ez-types.raml]: Done in 24b325688faeab5bf25cfebda159072385718518

**Review by adriendupuis — APPROVED** [review]

**Pull request merged**